include/swift/SIL/SILLocation.h

```diff
@@ -122,12 +123,20 @@ class SILLocation {
     SourceLoc SILFileLoc;
 
     /// A deserialized source location.
-    DebugLoc DebugInfoLoc;
+    struct CompactDebugLoc {
```

Comment:
CompactDebugLoc is a slightly misleading name.
SILLocation is a wrapper around different underlying storage mechanisms:
- ASTNodes
- SourceLoc
- DebugInfoLoc

These are not full interchangeable. You could *lower* ASTNode- and SourceLoc-backed SILLocations into DebugInfoLoc-backed ones, but then you may loose additional info that is needed to emit diagnostics. The primary motivation behind DebugInfoLoc is that we needed a way to represent SILLocations that are materialized from textual SIL, where no ASTNodes are available.

Reply:
This was a while back though and we are doing a lot more inline-like (in the sense of that they clone entire SILDebugScope trees) optimizations such as generic specialization now.

Reply:
"DecomposedDebugLoc", maybe? It's storing the same data as DebugLoc except for the filename length.

Reply:
Ah.. now I get it. It wasn't clear to me that DebugLoc literally referred to another type in the same file called `DebugLoc`. I think CompactDebugLoc makes sense then. (Perhaps add the DebugLoc \ Name length into the doxygen comment?).